lib/iris/_deprecation_helpers.py

```diff
@@ -24,10 +24,20 @@
 
 import warnings
 
+from iris.exceptions import IrisError
+
+
+class IrisDeprecation(IrisError, DeprecationWarning):
```

Comment:
> Perhaps we should have two classes

NB. I only mentioned this for completeness, not because I have any real appetite for two distinct classes!